sound/soc/sof/ipc4-topology.c

```diff
@@ -516,7 +533,8 @@ static void sof_ipc4_widget_free_comp_dai(struct snd_sof_widget *swidget)
 	kfree(available_fmt->dma_buffer_size);
 	kfree(available_fmt->base_config);
 	kfree(available_fmt->out_audio_fmt);
-	kfree(ipc4_copier->copier_config);
+	if (ipc4_copier->dai_type == SOF_DAI_INTEL_HDA)
+		kfree(ipc4_copier->copier_config);
```

Comment:
Should this be `if (ipc4_copier->dai_type != SOF_DAI_INTEL_SSP && ipc4_copier->dai_type != SOF_DAI_INTEL_DMIC)`? We allocate `ipc4_copier->copier_config` in the default case in `sof_ipc4_widget_setup_comp_dai`, so I think we should free `ipc4_copier->copier_config` in all dai_type except `SOF_DAI_INTEL_SSP` and `SOF_DAI_INTEL_DMIC`.
We will allocate ipc4_copier->copier_config in `SOF_DAI_INTEL_ALH` type, too.

Reply:
> Should this be `if (ipc4_copier->dai_type != SOF_DAI_INTEL_SSP && ipc4_copier->dai_type != SOF_DAI_INTEL_DMIC)`? We allocate `ipc4_copier->copier_config` in the default case in `sof_ipc4_widget_setup_comp_dai`, so I think we should free `ipc4_copier->copier_config` in all dai_type except `SOF_DAI_INTEL_SSP` and `SOF_DAI_INTEL_DMIC`. We will allocate ipc4_copier->copier_config in `SOF_DAI_INTEL_ALH` type, too.

Chat with @ranj063, I will merge it now and fix it with a fixup commit,